config/chassis_modules.py

```diff
@@ -17,6 +22,81 @@ def modules():
     """Configure chassis modules"""
     pass
 
+
+def get_config_module_state(db, chassis_module_name):
+    config_db = db.cfgdb
+    fvs = config_db.get_entry('CHASSIS_MODULE', chassis_module_name)
+    if not fvs:
+        return 'up'
+    else:
+        return fvs['admin_status']
+
+
+#
+# Name: check_config_module_state_with_timeout
+# return: True: timeout, False: not timeout
+#
+def check_config_module_state_with_timeout(ctx, db, chassis_module_name, state):
+    counter = 0
+    while get_config_module_state(db, chassis_module_name) != state:
+        time.sleep(1)
+        counter += 1
+        if counter >= TIMEOUT_SECS:
+            ctx.fail("get_config_module_state {} timeout".format(chassis_module_name))
+            return True
+    return False
+
+
+def get_asic_list_from_db(chassisdb, chassis_module_name):
+    asic_list = []
+    asics_keys_list = chassisdb.keys("CHASSIS_STATE_DB", "CHASSIS_FABRIC_ASIC_TABLE*")
+    for asic_key in asics_keys_list:
+        name = chassisdb.get("CHASSIS_STATE_DB", asic_key, "name")
+        if name == chassis_module_name:
+            asic_id = int(re.search(r"(\d+)$", asic_key).group())
+            asic_list.append(asic_id)
+    return asic_list
+
+
+#
+# Syntax: fabric_module_set_admin_status <chassis_module_name> <'up'/'down'>
+#
+def fabric_module_set_admin_status(db, chassis_module_name, state):
+    chassisdb = db.db
+    chassisdb.connect("CHASSIS_STATE_DB")
+    asic_list = get_asic_list_from_db(chassisdb, chassis_module_name)
+
+    if len(asic_list) == 0:
+        return
+
+    if state == "down":
+        for asic in asic_list:
+            click.echo("Stop swss@{} and peer services".format(asic))
+            clicommon.run_command('sudo systemctl stop swss@{}.service'.format(asic))
+
+        is_active = subprocess.call(["systemctl", "is-active", "--quiet", "swss@{}.service".format(asic)])
+
+        if is_active == 0:  # zero active,  non-zero, inactive
+            click.echo("Stop swss@{} and peer services failed".format(asic))
+            return
+
+        click.echo("Delete related CAHSSIS_FABRIC_ASIC_TABLE entries")
+
+        for asic in asic_list:
+            chassisdb.delete("CHASSIS_STATE_DB", "CHASSIS_FABRIC_ASIC_TABLE|asic" + str(asic))
+
+        # Start the services in case of the users just execute issue command "systemctl stop swss@/syncd@"
+        # without bring down the hardware
+        for asic in asic_list:
+            click.echo("Start swss@{} and peer services".format(asic))
```

Comment:
Nitpick: the string does not coincide with the subsequent command; e.g. `sudo systemctl reset-failed`. Please revise accordingly

Reply:
Addressed.